cluster/gce/gci/configure-helper_test.go

```diff
@@ -0,0 +1,215 @@
+readonly KUBE_API_SERVER_DOCKER_TAG=v1.11.0-alpha.0.1808_3c7452dc11645d-dirty
+readonly LOG_OWNER_USER=$(whoami)
+readonly LOG_OWNER_GROUP=$(id -gn)
+ENCRYPTION_PROVIDER_CONFIG={{.EncryptionProviderConfig}}
+ENCRYPTION_PROVIDER_CONFIG_PATH={{.EncryptionProviderConfigPath}}
+readonly ETCD_KMS_KEY_ID={{.ETCDKMSKeyID}}
+`
+	kubeApiServerManifestFileName = "kube-apiserver.manifest"
+	kmsPluginManifestFileName     = "kms-plugin-container.manifest"
+	kubeApiServerStartFuncName    = "start-kube-apiserver"
+
+	// Position of containers within a pod manifest
+	kmsPluginContainerIndex        = 0
+	apiServerContainerIndexNoKMS   = 0
+	apiServerContainerIndexWithKMS = 1
+
+	//	command": [
+	//   "/bin/sh", - Index 0
+	//   "-c",      - Index 1
+	//   "exec /usr/local/bin/kube-apiserver " - Index 2
+	execArgsIndex = 2
+
+	socketVolumeMountIndexKMSPlugin = 1
+	socketVolumeMountIndexApiServer = 0
+)
+
+type kubeApiServerEnv struct {
+	KubeHome                     string
+	EncryptionProviderConfig     string
+	EncryptionProviderConfigPath string
+	ETCDKMSKeyID                 string
+}
+
+type kubeApiServerManifestTestCase struct {
+	*ManifestTestCase
+	apiServerContainer v1.Container
+	kmsPluginContainer v1.Container
+}
+
+func NewKubeApiServerManifestTestCase(t *testing.T) *kubeApiServerManifestTestCase {
+	c := &kubeApiServerManifestTestCase{}
+	c.ManifestTestCase = NewManifestTestCase(kubeApiServerManifestFileName, kubeApiServerStartFuncName, []string{kmsPluginManifestFileName}, t)
+
+	return c
+}
+
+func (c *kubeApiServerManifestTestCase) getContainers() {
+	if len(c.pod.Spec.Containers) == 0 {
+		c.t.Fatal("Expected at least one container in apiserver pod, but got 0.")
+	}
+
+	if len(c.pod.Spec.Containers) > 2 {
+		c.t.Fatalf("Expected at most two containers in apiserver pod, but got %d.", len(c.pod.Spec.Containers))
+	}
+
+	if len(c.pod.Spec.Containers) == 2 {
+		c.apiServerContainer = c.pod.Spec.Containers[apiServerContainerIndexWithKMS]
+		c.kmsPluginContainer = c.pod.Spec.Containers[kmsPluginContainerIndex]
+	} else {
+		c.apiServerContainer = c.pod.Spec.Containers[apiServerContainerIndexNoKMS]
+	}
+}
+
+func (c *kubeApiServerManifestTestCase) invokeTest(e kubeApiServerEnv) {
+	c.MustInvokeFunc(deployHelperEnv, e)
+	c.MustLoadPodFromManifest()
+	c.getContainers()
+}
+
+func TestEncryptionProviderFlag(t *testing.T) {
+	c := NewKubeApiServerManifestTestCase(t)
+	defer c.TearDown()
+
+	var e = kubeApiServerEnv{
+		KubeHome:                     c.kubeHome,
+		EncryptionProviderConfig:     base64.StdEncoding.EncodeToString([]byte("FOO")),
+		EncryptionProviderConfigPath: filepath.Join(c.kubeHome, "encryption-provider-config.yaml"),
+	}
+
+	c.invokeTest(e)
+
+	expectedFlag := fmt.Sprintf("--experimental-encryption-provider-config=%s", e.EncryptionProviderConfigPath)
+	execArgs := c.apiServerContainer.Command[execArgsIndex]
+	if !strings.Contains(execArgs, expectedFlag) {
+		c.t.Fatalf("Expected %s flag, but did not find it in %s", expectedFlag, execArgs)
```

Comment:
nit: use %q for non-constant strings, to catch whitespace and invisible characters.

Reply:
Done.